web3/_utils/threads.py

```diff
@@ -60,16 +76,16 @@ def check(self):
             else:
                 raise self
 
-    def cancel(self):
+    def cancel(self) -> None:
         self.is_running = False
 
-    def sleep(self, seconds):
+    def sleep(self, seconds: float) -> None:
         time.sleep(seconds)
         self.check()
 
 
 class ThreadWithReturn(threading.Thread):
```

Comment:
I think that this should be typed with a `Generic[TReturn]`.  This allows you to then have the type of `target: Callable[..., TReturn`, and thus to be able to provide the type for the `def get(self, ...) -> TReturn`